src/renderer/components/message/MessageList.tsx

```diff
@@ -644,7 +644,8 @@ export const MessageListInner = React.memo(
       prevProps.messageCache === nextProps.messageCache &&
       prevProps.oldestFetchedMessageIndex ===
         nextProps.oldestFetchedMessageIndex &&
-      prevProps.onScroll === nextProps.onScroll
+      prevProps.onScroll === nextProps.onScroll &&
+      prevProps.messageListItems.length === prevProps.messageListItems.length
```

Comment:
I guess this will fix the particular bug, but, as you mentioned, I believe there is a bigger underlying problem with `memo`.
It was added here https://github.com/deltachat/deltachat-desktop/pull/2535 and was just a way to save memory, if I understand it correctly. After #2987 we are no longer using pages, but messageCache, which is id-addressed, so I suspect we no longer need this. Either way, the manual `memo` condition checking doesn't seem to be saving us too many re-renders - I tried to do a few things and it re-renders practically every time something happens. Perhaps need to just re-render it when any of the props change.

Reply:
Let's already merge this PR and discuss about it later.

Reply:
Is the problem fixed if the `React.memo()` is removed altogether?

Reply:
It should, though I did not check.
I'd keep `React.memo` and only remove `arePropsEqual`, but need to check whether it really is not needed anymore.

Reply:
Removed manual check